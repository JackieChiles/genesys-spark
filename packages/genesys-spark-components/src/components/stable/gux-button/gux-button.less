@import (reference) '../../../style/color.less';
@import (reference) '../../../style/mixins.less';
@import (reference) '../../../style/typography.less';
@import (reference) '../../../style/spacing.less';
@import (reference) '../../../style/color-mixins.less';

:host {
  display: inline-block;
  pointer-events: none;
  -webkit-user-select: none;
  user-select: none;
}

:host(:focus) {
  outline: none;

  button {
    outline: none;
  }
}

::slotted(gux-icon) {
  width: 16px;
  height: 16px;
}

::slotted(*) {
  padding-left: @gux-spacing-2xs;
  vertical-align: middle;
}

::slotted(*:first-child) {
  padding: 0;
}

button {
  width: 100%;
  min-width: 32px;
  height: 32px;
  padding: 0 @gux-spacing-medium;
  overflow: hidden;
  color: @gux-black-50;
  text-overflow: ellipsis;
  white-space: nowrap;
  pointer-events: auto;
  cursor: pointer;
  background-color: @gux-grey-60;
  border: none;
  border-radius: 4px;
  .body-font-bold();

  &[disabled] {
    color: fade(@gux-black-50, 50%);
    cursor: default;
    background-color: fade(@gux-grey-60, 50%);
  }

  &:hover:enabled {
    background-color: @gux-grey-50;
  }

  &:active:enabled {
    background-color: @gux-grey-40;
  }

  &:focus:enabled {
    .focus-ring();
  }

  &.gux-tertiary {
    color: @gux-blue-60;
    background-color: @gux-grey-100;
    border: 1px solid @gux-blue-60;

    &[disabled] {
      color: fade(@gux-blue-60, 50%);
      background-color: fade(@gux-grey-100, 50%);
      border-color: fade(@gux-blue-60, 50%);
    }

    &:hover:enabled {
      color: @gux-grey-100;
      background-color: @gux-blue-50;
    }

    &:active:enabled {
      color: @gux-grey-100;
      background-color: @gux-blue-40;
    }
  }

  &.gux-ghost {
    color: @gux-blue-60;
    background: none;
    border: none;

    &[disabled] {
      color: fade(@gux-blue-60, 50%);
    }

    &:hover:enabled {
      color: @gux-blue-50;
      background-color: fade(@gux-blue-100, 33%);
    }

    &:active:enabled {
      color: @gux-blue-50;
      background-color: @gux-blue-100;
    }
  }

  &.gux-primary {
    .gux-blue-60();

    &[disabled] {
      color: fade(@gux-grey-100, 50%);
      background-color: fade(@gux-blue-60, 50%);
    }

    &:hover:enabled {
      background-color: @gux-blue-50;
    }

    &:active:enabled {
      background-color: @gux-blue-40;
    }
  }

  &.gux-danger {
    .gux-alert-red-60();

    &[disabled] {
      color: fade(@gux-grey-100, 50%);
      background-color: fade(@gux-alert-red-60, 50%);
    }

    &:hover:enabled {
      background-color: @gux-alert-red-50;
    }

    &:active:enabled {
      background-color: @gux-alert-red-40;
    }
  }

  &.gux-inline {
    min-width: initial;
    height: initial;
    padding: 0;
    color: @gux-blue-60;
    text-decoration: underline;
    background: none;
    border: none;
    .body-font();

    &[disabled] {
      color: fade(@gux-blue-60, 50%);
    }

    &:hover:enabled {
      color: @gux-blue-50;
      background-color: fade(@gux-blue-100, 33%);
    }

    &:active:enabled {
      color: @gux-blue-50;
      background-color: @gux-blue-100;
    }
  }
}
