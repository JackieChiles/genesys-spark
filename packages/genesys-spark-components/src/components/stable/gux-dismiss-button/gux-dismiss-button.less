@import (reference) '../../../style/color.less';
@import (reference) '../../../style/spacing.less';
@import (reference) '../../../style/focus.less';

button {
  position: absolute;
  top: 0;
  right: 0;
  padding: 2px;
  color: @gux-black-90;
  background: transparent;
  border: none;
  border-radius: 4px;

  &.gux-inherit {
    position: inherit;
  }

  &:not(:disabled):focus-visible,
  &:not(:disabled):hover {
    color: @gux-black-50;
    cursor: pointer;
  }

  gux-icon {
    width: 16px;
    height: 16px;
    margin: @gux-spacing-2xs;
    border-radius: 4px;
  }

  &:focus {
    outline: none;

    gux-icon {
      outline: none;
    }
  }

  &:focus-visible:enabled {
    gux-icon {
      .gux-focus-ring();
    }
  }
}
