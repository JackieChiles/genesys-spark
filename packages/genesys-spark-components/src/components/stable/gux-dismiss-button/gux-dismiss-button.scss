:host {
  // all these to be removed when token issues are fixed
  --gse-ui-button-dismiss-medium-width: 32px;
  --gse-ui-button-dismiss-medium-height: 32px;
  --gse-ui-button-dismiss-small-width: 24px;
  --gse-ui-button-dismiss-small-height: 24px;
}

gux-button-slot {
  position: absolute;
  top: var(--gse-ui-modal-dismissButton-paddingTop);
  right: var(--gse-ui-modal-dismissButton-paddingRight);
  width: var(--gse-ui-button-dismiss-medium-width);
  height: var(--gse-ui-button-dismiss-medium-height);

  &.gux-dismiss-small {
    width: var(--gse-ui-button-dismiss-small-width);
    height: var(--gse-ui-button-dismiss-small-height);

    button {
      width: var(--gse-ui-button-dismiss-small-width);
      height: var(
        --gse-ui-button-dismiss-small-height
      ); // button doesn't stay the same height for the small one without this

      padding: 0;
    }
  }

  &.gux-inherit {
    position: inherit;
  }

  .gux-icon-container {
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    place-content: stretch center;
    align-items: center;

    gux-icon {
      flex: 0 0 auto;
      align-self: auto;
      order: 0;
      width: 16px;
      height: 16px;
      color: var(--gse-ui-dismissButton-foregroundColor);
    }
  }
}
