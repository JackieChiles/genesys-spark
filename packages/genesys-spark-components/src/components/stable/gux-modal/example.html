<h1>gux-modal</h1>

<gux-modal is-open id="example1">
  <div slot="title">Dialog Title 1</div>
  <div slot="content">This contains the dialog content</div>
  <div slot="start-align-buttons">
    <gux-button type="button" accent="primary">Accept</gux-button>
  </div>
  <div slot="end-align-buttons">
    <gux-button
      onclick="document.getElementById('example1').hideModal()"
      id="cancel-button"
      type="button"
      >Cancel</gux-button
    >
  </div>
</gux-modal>
<gux-button onclick="document.getElementById('example1').showModal()"
  >Show</gux-button
>

<gux-modal id="example2">
  <div slot="title">Dialog Title 2</div>
  <div slot="content">This contains the dialog content</div>
  <div slot="start-align-buttons">
    <gux-button-slot accent="primary">
      <button autofocus type="button">Accept</button>
    </gux-button-slot>
  </div>
  <div slot="end-align-buttons">
    <gux-button-slot>
      <button
        onclick="document.getElementById('example2').hideModal()"
        type="button"
      >
        Cancel
      </button>
    </gux-button-slot>
  </div>
</gux-modal>
<gux-button onclick="document.getElementById('example2').showModal()"
  >Show</gux-button
>
