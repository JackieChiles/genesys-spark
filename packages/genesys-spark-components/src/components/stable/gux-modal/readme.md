# gux-modal

<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description                                              | Type                                          | Default     |
| -------- | --------- | -------------------------------------------------------- | --------------------------------------------- | ----------- |
| `size`   | `size`    | Indicates the size of the modal (small, medium or large) | `"dynamic" \| "large" \| "medium" \| "small"` | `'dynamic'` |


## Events

| Event        | Description                           | Type                |
| ------------ | ------------------------------------- | ------------------- |
| `guxdismiss` | Fired when a user dismisses the modal | `CustomEvent<void>` |


## Methods

### `hideModal() => Promise<void>`



#### Returns

Type: `Promise<void>`



### `showModal() => Promise<void>`



#### Returns

Type: `Promise<void>`




## Dependencies

### Depends on

- [gux-dismiss-button](../gux-dismiss-button)

### Graph
```mermaid
graph TD;
  gux-modal --> gux-dismiss-button
  gux-dismiss-button --> gux-button-slot
  gux-dismiss-button --> gux-icon
  style gux-modal fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
