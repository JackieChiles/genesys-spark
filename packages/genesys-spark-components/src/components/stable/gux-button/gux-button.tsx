import { Component, Element, h, JSX, Prop, State } from '@stencil/core';

import { trackComponent } from '@utils/tracking/usage';
import { GuxButtonAccent, GuxButtonType } from './gux-button.types';

/**
 * @slot - content
 */

@Component({
  styleUrl: 'gux-button.scss',
  tag: 'gux-button',
  shadow: { delegatesFocus: true }
})
export class GuxButton {
  @Element()
  private root: HTMLElement;

  /**
   * The component button type
   */
  @Prop()
  type: GuxButtonType = 'button';

  /**
   * The component title
   */
  @Prop()
  guxTitle: string;

  /**
   * Indicate if the button is disabled or not
   */
  @Prop()
  disabled: boolean = false;

  @Prop()
  accent: GuxButtonAccent = 'secondary';

  @State()
  iconOnly: boolean;

  connectedCallback() {
    this.slotChanged();
  }

  componentWillLoad() {
    trackComponent(this.root, { variant: this.accent });
  }

  render(): JSX.Element {
    return (
      <button
        type={this.type}
        title={this.guxTitle}
        disabled={this.disabled}
        class={{
          [`gux-${this.accent}`]: true,
          'gux-icon-only': this.iconOnly
        }}
      >
        <slot onSlotchange={this.slotChanged.bind(this)} />
      </button>
    ) as JSX.Element;
  }

  private stopEventIfDisabled(event: Event) {
    if (this.disabled) {
      event.stopImmediatePropagation();
      event.stopPropagation();
      event.preventDefault();
    }
  }

  private makeSlotContentDisableable() {
    this.root.shadowRoot.addEventListener('click', (event: MouseEvent) =>
      this.stopEventIfDisabled(event)
    );

    Array.from(this.root.children).forEach(slotElement => {
      slotElement.addEventListener('click', (event: MouseEvent) =>
        this.stopEventIfDisabled(event)
      );
    });
  }

  private hasIconOnly(): boolean {
    const children = Array.from(this.root.children);

    if (children.length === 1) {
      const child = children[0];
      if (child.tagName === 'GUX-ICON') {
        return true;
      }
    }

    return false;
  }

  private slotChanged() {
    this.makeSlotContentDisableable();
    this.iconOnly = this.hasIconOnly();
  }
}
