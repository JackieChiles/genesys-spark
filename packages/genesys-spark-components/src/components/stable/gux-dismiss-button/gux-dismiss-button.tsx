import { Component, Element, h, JSX, Prop } from '@stencil/core';

import { trackComponent } from '@utils/tracking/usage';

import { buildI18nForComponent, GetI18nValue } from '../../../i18n';

import translationResources from './i18n/en.json';
import { GuxDismissButtonPosition } from './gux-dismiss-button.types';

@Component({
  styleUrl: 'gux-dismiss-button.less',
  tag: 'gux-dismiss-button',
  shadow: { delegatesFocus: true }
})
export class GuxDismissButton {
  private i18n: GetI18nValue;

  @Element()
  private root: HTMLElement;

  @Prop()
  position: GuxDismissButtonPosition = 'absolute';

  async componentWillLoad(): Promise<void> {
    trackComponent(this.root, { variant: this.position });
    this.i18n = await buildI18nForComponent(this.root, translationResources);
  }

  render(): JSX.Element {
    return (
      <button
        class={this.position == 'inherit' ? 'gux-inherit' : undefined}
        type="button"
        title={this.i18n('dismiss')}
      >
        <gux-icon
          icon-name="close"
          screenreader-text={this.i18n('dismiss')}
        ></gux-icon>
      </button>
    ) as JSX.Element;
  }
}
