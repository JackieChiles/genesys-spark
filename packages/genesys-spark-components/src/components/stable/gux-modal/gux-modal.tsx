import {
  Component,
  Element,
  Event,
  EventEmitter,
  h,
  JSX,
  Prop,
  Method
} from '@stencil/core';

import { trackComponent } from '@utils/tracking/usage';
import { GuxModalSize } from './gux-modal.types';

@Component({
  styleUrl: 'gux-modal.scss',
  tag: 'gux-modal',
  shadow: true
})
export class GuxModal {
  private dialogElement: HTMLDialogElement;

  @Element()
  private root: HTMLElement;

  /**
   * Indicates if the modal is initially shown
   */
  @Prop()
  isOpen: boolean = false;

  /**
   * Indicates the size of the modal (small, medium or large)
   */
  // not yet implemented
  @Prop()
  size: GuxModalSize = 'dynamic';

  /**
   * Query selector for the element to initially focus when the modal opens
   * Defaults to the first tabbable element
   */
  // not yet implemented
  @Prop()
  initialFocus?: string | undefined;

  /**
   * Fired when a user dismisses the modal
   */
  @Event()
  guxdismiss: EventEmitter<void>;

  // eslint-disable-next-line @typescript-eslint/require-await
  @Method()
  async showModal(): Promise<void> {
    this.dialogElement.showModal();
  }

  // eslint-disable-next-line @typescript-eslint/require-await
  @Method()
  async hideModal(): Promise<void> {
    // this.guxdismiss.emit();
    this.dialogElement.close();
  }

  componentWillLoad(): void {
    trackComponent(this.root, { variant: `${this.size}` });
  }

  componentDidLoad(): void {
    if (this.isOpen) {
      this.dialogElement.showModal();
    }
  }

  render(): JSX.Element {
    return (
      <dialog
        onClose={this.onCloseHandler.bind(this)}
        ref={el => (this.dialogElement = el)}
      >
        <div class={`gux-modal-container gux-${this.size}`}>
          <gux-dismiss-button
            onClick={this.onDismissHandler.bind(this)}
          ></gux-dismiss-button>
          <h1 class="gux-modal-header">
            <slot name="title" />
          </h1>
          <slot name="content" />
          <div class="gux-button-footer">
            <div class="gux-start-align-buttons">
              <slot name="start-align-buttons" />
            </div>

            <div class="gux-end-align-buttons">
              <slot name="end-align-buttons" />
            </div>
          </div>
        </div>
      </dialog>
    ) as JSX.Element;
  }

  private onCloseHandler(): void {
    this.guxdismiss.emit();
  }

  private onDismissHandler(): void {
    this.dialogElement.close();
  }
}
