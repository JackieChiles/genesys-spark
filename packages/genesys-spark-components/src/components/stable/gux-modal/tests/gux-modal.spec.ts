import { newSpecPage } from '@test/specTestUtils';
import { GuxButton } from '../../gux-button/gux-button';
import { GuxModal } from '../gux-modal';
import { GuxDismissButton } from '../../gux-dismiss-button/gux-dismiss-button';

const components = [GuxButton, GuxModal, GuxDismissButton];
const language = 'en';

describe('gux-modal', () => {
  describe('#render', () => {
    [
      {
        description: 'should render small modal',
        html: `
          <gux-modal size="small">
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
            <div slot="start-align-buttons">
                <gux-button-slot><button>Cancel</button></gux-button-slot>
            </div>
            <div slot="end-align-buttons">
              <gux-button-slot accent='primary'><button>Accept</button></gux-button-slot>
            </div>
          </gux-modal>
        `
      },
      {
        description: 'should render medium modal',
        html: `
          <gux-modal size="medium">
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
            <div slot="start-align-buttons">
                <gux-button-slot><button>Cancel</button></gux-button-slot>
            </div>
            <div slot="end-align-buttons">
              <gux-button-slot accent='primary'><button>Accept</button></gux-button-slot>
            </div>
          </gux-modal>
        `
      },
      {
        description: 'should render large modal',
        html: `
          <gux-modal size="large">
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
            <div slot="start-align-buttons">
                <gux-button-slot><button>Cancel</button></gux-button-slot>
            </div>
            <div slot="end-align-buttons">
              <gux-button-slot accent='primary'><button>Accept</button></gux-button-slot>
            </div>
          </gux-modal>
        `
      },
      {
        description: 'should render modal without a title',
        html: `
          <gux-modal size="large">
            <div slot="content">This contains the modal content.</div>
            <div slot="start-align-buttons">
                <gux-button-slot><button>Cancel</button></gux-button-slot>
            </div>
            <div slot="end-align-buttons">
              <gux-button-slot accent='primary'><button>Accept</button></gux-button-slot>
            </div>
          </gux-modal>
        `
      },
      {
        description: 'should render modal without buttons',
        html: `
          <gux-modal size="small">
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
          </gux-modal>
        `
      },
      {
        description: 'should render modal with just left align buttons',
        html: `
          <gux-modal size="small">
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
            <div slot="start-align-buttons">
                <gux-button-slot><button>Cancel</button></gux-button-slot>
            </div>
          </gux-modal>
        `
      },
      {
        description: 'should render modal with just right align buttons',
        html: `
          <gux-modal size="small">
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
            <div slot="end-align-buttons">
              <gux-button-slot accent='primary'><button>Accept</button></gux-button-slot>
            </div>
          </gux-modal>
        `
      },
      {
        description:
          'should render modal with a specified initial focus element',
        html: `
          <gux-modal initial-focus="#cancelButton">
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
            <div slot="start-align-buttons">
                <gux-button id="cancelButton">Cancel</gux-button>
            </div>
            <div slot="end-align-buttons">
              <gux-button-slot accent='primary'><button>Accept</button></gux-button-slot>
            </div>
          </gux-modal>
        `
      },
      {
        description: 'should render small modal by default',
        html: `
          <gux-modal>
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
            <div slot="start-align-buttons">
                <gux-button-slot><button>Cancel</button></gux-button-slot>
            </div>
            <div slot="end-align-buttons">
              <gux-button-slot accent='primary'><button>Accept</button></gux-button-slot>
            </div>
          </gux-modal>
        `
      }
    ].forEach(({ description, html }) => {
      it(description, async () => {
        const page = await newSpecPage({ components, html, language });

        expect(page.rootInstance).toBeInstanceOf(GuxModal);

        expect(page.root).toMatchSnapshot();
      });
    });
  });
});
