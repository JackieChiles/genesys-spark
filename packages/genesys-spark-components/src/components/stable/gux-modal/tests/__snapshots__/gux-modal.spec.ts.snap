// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-modal #render should render large modal 1`] = `
<gux-modal size="large">
  <template shadowrootmode="open">
    <dialog aria-labelledby="gux-i">
      <div class="gux-large gux-modal-container">
        <gux-dismiss-button>
          <template shadowrootmode="open">
            <gux-button-slot accent="ghost">
              <button title="Dismiss" type="button">
                <div class="gux-icon-container">
                  <gux-icon decorative="" icon-name="fa/xmark-large-regular"></gux-icon>
                </div>
              </button>
            </gux-button-slot>
          </template>
        </gux-dismiss-button>
        <h1 class="gux-modal-header" id="gux-i">
          <slot name="title"></slot>
        </h1>
        <div class="gux-modal-content">
          <p>
            <slot name="content"></slot>
          </p>
        </div>
        <div class="gux-button-footer">
          <div class="gux-start-align-buttons">
            <slot name="start-align-buttons"></slot>
          </div>
          <div class="gux-end-align-buttons">
            <slot name="end-align-buttons"></slot>
          </div>
        </div>
      </div>
    </dialog>
  </template>
  <div slot="title">
    Modal Title
  </div>
  <div slot="content">
    This contains the modal content.
  </div>
  <div slot="start-align-buttons">
    <gux-button-slot>
      <button>
        Cancel
      </button>
    </gux-button-slot>
  </div>
  <div slot="end-align-buttons">
    <gux-button-slot accent="primary">
      <button>
        Accept
      </button>
    </gux-button-slot>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render medium modal 1`] = `
<gux-modal size="medium">
  <template shadowrootmode="open">
    <dialog aria-labelledby="gux-i">
      <div class="gux-medium gux-modal-container">
        <gux-dismiss-button>
          <template shadowrootmode="open">
            <gux-button-slot accent="ghost">
              <button title="Dismiss" type="button">
                <div class="gux-icon-container">
                  <gux-icon decorative="" icon-name="fa/xmark-large-regular"></gux-icon>
                </div>
              </button>
            </gux-button-slot>
          </template>
        </gux-dismiss-button>
        <h1 class="gux-modal-header" id="gux-i">
          <slot name="title"></slot>
        </h1>
        <div class="gux-modal-content">
          <p>
            <slot name="content"></slot>
          </p>
        </div>
        <div class="gux-button-footer">
          <div class="gux-start-align-buttons">
            <slot name="start-align-buttons"></slot>
          </div>
          <div class="gux-end-align-buttons">
            <slot name="end-align-buttons"></slot>
          </div>
        </div>
      </div>
    </dialog>
  </template>
  <div slot="title">
    Modal Title
  </div>
  <div slot="content">
    This contains the modal content.
  </div>
  <div slot="start-align-buttons">
    <gux-button-slot>
      <button>
        Cancel
      </button>
    </gux-button-slot>
  </div>
  <div slot="end-align-buttons">
    <gux-button-slot accent="primary">
      <button>
        Accept
      </button>
    </gux-button-slot>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render modal with a specified initial focus element 1`] = `
<gux-modal initial-focus="#cancelButton">
  <template shadowrootmode="open">
    <dialog aria-labelledby="gux-i">
      <div class="gux-dynamic gux-modal-container">
        <gux-dismiss-button>
          <template shadowrootmode="open">
            <gux-button-slot accent="ghost">
              <button title="Dismiss" type="button">
                <div class="gux-icon-container">
                  <gux-icon decorative="" icon-name="fa/xmark-large-regular"></gux-icon>
                </div>
              </button>
            </gux-button-slot>
          </template>
        </gux-dismiss-button>
        <h1 class="gux-modal-header" id="gux-i">
          <slot name="title"></slot>
        </h1>
        <div class="gux-modal-content">
          <p>
            <slot name="content"></slot>
          </p>
        </div>
        <div class="gux-button-footer">
          <div class="gux-start-align-buttons">
            <slot name="start-align-buttons"></slot>
          </div>
          <div class="gux-end-align-buttons">
            <slot name="end-align-buttons"></slot>
          </div>
        </div>
      </div>
    </dialog>
  </template>
  <div slot="title">
    Modal Title
  </div>
  <div slot="content">
    This contains the modal content.
  </div>
  <div slot="start-align-buttons">
    <gux-button id="cancelButton">
      <template shadowrootmode="open">
        <button class="gux-secondary" type="button">
          <slot></slot>
        </button>
      </template>
      Cancel
    </gux-button>
  </div>
  <div slot="end-align-buttons">
    <gux-button-slot accent="primary">
      <button>
        Accept
      </button>
    </gux-button-slot>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render modal with just left align buttons 1`] = `
<gux-modal size="small">
  <template shadowrootmode="open">
    <dialog aria-labelledby="gux-i">
      <div class="gux-modal-container gux-small">
        <gux-dismiss-button>
          <template shadowrootmode="open">
            <gux-button-slot accent="ghost">
              <button title="Dismiss" type="button">
                <div class="gux-icon-container">
                  <gux-icon decorative="" icon-name="fa/xmark-large-regular"></gux-icon>
                </div>
              </button>
            </gux-button-slot>
          </template>
        </gux-dismiss-button>
        <h1 class="gux-modal-header" id="gux-i">
          <slot name="title"></slot>
        </h1>
        <div class="gux-modal-content">
          <p>
            <slot name="content"></slot>
          </p>
        </div>
        <div class="gux-button-footer">
          <div class="gux-start-align-buttons">
            <slot name="start-align-buttons"></slot>
          </div>
          <div class="gux-end-align-buttons">
            <slot name="end-align-buttons"></slot>
          </div>
        </div>
      </div>
    </dialog>
  </template>
  <div slot="title">
    Modal Title
  </div>
  <div slot="content">
    This contains the modal content.
  </div>
  <div slot="start-align-buttons">
    <gux-button-slot>
      <button>
        Cancel
      </button>
    </gux-button-slot>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render modal with just right align buttons 1`] = `
<gux-modal size="small">
  <template shadowrootmode="open">
    <dialog aria-labelledby="gux-i">
      <div class="gux-modal-container gux-small">
        <gux-dismiss-button>
          <template shadowrootmode="open">
            <gux-button-slot accent="ghost">
              <button title="Dismiss" type="button">
                <div class="gux-icon-container">
                  <gux-icon decorative="" icon-name="fa/xmark-large-regular"></gux-icon>
                </div>
              </button>
            </gux-button-slot>
          </template>
        </gux-dismiss-button>
        <h1 class="gux-modal-header" id="gux-i">
          <slot name="title"></slot>
        </h1>
        <div class="gux-modal-content">
          <p>
            <slot name="content"></slot>
          </p>
        </div>
        <div class="gux-button-footer">
          <div class="gux-start-align-buttons">
            <slot name="start-align-buttons"></slot>
          </div>
          <div class="gux-end-align-buttons">
            <slot name="end-align-buttons"></slot>
          </div>
        </div>
      </div>
    </dialog>
  </template>
  <div slot="title">
    Modal Title
  </div>
  <div slot="content">
    This contains the modal content.
  </div>
  <div slot="end-align-buttons">
    <gux-button-slot accent="primary">
      <button>
        Accept
      </button>
    </gux-button-slot>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render modal without a title 1`] = `
<gux-modal size="large">
  <template shadowrootmode="open">
    <dialog>
      <div class="gux-large gux-modal-container">
        <gux-dismiss-button>
          <template shadowrootmode="open">
            <gux-button-slot accent="ghost">
              <button title="Dismiss" type="button">
                <div class="gux-icon-container">
                  <gux-icon decorative="" icon-name="fa/xmark-large-regular"></gux-icon>
                </div>
              </button>
            </gux-button-slot>
          </template>
        </gux-dismiss-button>
        <div class="gux-modal-content">
          <p>
            <slot name="content"></slot>
          </p>
        </div>
        <div class="gux-button-footer">
          <div class="gux-start-align-buttons">
            <slot name="start-align-buttons"></slot>
          </div>
          <div class="gux-end-align-buttons">
            <slot name="end-align-buttons"></slot>
          </div>
        </div>
      </div>
    </dialog>
  </template>
  <div slot="content">
    This contains the modal content.
  </div>
  <div slot="start-align-buttons">
    <gux-button-slot>
      <button>
        Cancel
      </button>
    </gux-button-slot>
  </div>
  <div slot="end-align-buttons">
    <gux-button-slot accent="primary">
      <button>
        Accept
      </button>
    </gux-button-slot>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render modal without buttons 1`] = `
<gux-modal size="small">
  <template shadowrootmode="open">
    <dialog aria-labelledby="gux-i">
      <div class="gux-modal-container gux-small">
        <gux-dismiss-button>
          <template shadowrootmode="open">
            <gux-button-slot accent="ghost">
              <button title="Dismiss" type="button">
                <div class="gux-icon-container">
                  <gux-icon decorative="" icon-name="fa/xmark-large-regular"></gux-icon>
                </div>
              </button>
            </gux-button-slot>
          </template>
        </gux-dismiss-button>
        <h1 class="gux-modal-header" id="gux-i">
          <slot name="title"></slot>
        </h1>
        <div class="gux-modal-content">
          <p>
            <slot name="content"></slot>
          </p>
        </div>
        <div class="gux-button-footer gux-no-buttons">
          <div class="gux-start-align-buttons">
            <slot name="start-align-buttons"></slot>
          </div>
          <div class="gux-end-align-buttons">
            <slot name="end-align-buttons"></slot>
          </div>
        </div>
      </div>
    </dialog>
  </template>
  <div slot="title">
    Modal Title
  </div>
  <div slot="content">
    This contains the modal content.
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render small modal 1`] = `
<gux-modal size="small">
  <template shadowrootmode="open">
    <dialog aria-labelledby="gux-i">
      <div class="gux-modal-container gux-small">
        <gux-dismiss-button>
          <template shadowrootmode="open">
            <gux-button-slot accent="ghost">
              <button title="Dismiss" type="button">
                <div class="gux-icon-container">
                  <gux-icon decorative="" icon-name="fa/xmark-large-regular"></gux-icon>
                </div>
              </button>
            </gux-button-slot>
          </template>
        </gux-dismiss-button>
        <h1 class="gux-modal-header" id="gux-i">
          <slot name="title"></slot>
        </h1>
        <div class="gux-modal-content">
          <p>
            <slot name="content"></slot>
          </p>
        </div>
        <div class="gux-button-footer">
          <div class="gux-start-align-buttons">
            <slot name="start-align-buttons"></slot>
          </div>
          <div class="gux-end-align-buttons">
            <slot name="end-align-buttons"></slot>
          </div>
        </div>
      </div>
    </dialog>
  </template>
  <div slot="title">
    Modal Title
  </div>
  <div slot="content">
    This contains the modal content.
  </div>
  <div slot="start-align-buttons">
    <gux-button-slot>
      <button>
        Cancel
      </button>
    </gux-button-slot>
  </div>
  <div slot="end-align-buttons">
    <gux-button-slot accent="primary">
      <button>
        Accept
      </button>
    </gux-button-slot>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render small modal by default 1`] = `
<gux-modal>
  <template shadowrootmode="open">
    <dialog aria-labelledby="gux-i">
      <div class="gux-dynamic gux-modal-container">
        <gux-dismiss-button>
          <template shadowrootmode="open">
            <gux-button-slot accent="ghost">
              <button title="Dismiss" type="button">
                <div class="gux-icon-container">
                  <gux-icon decorative="" icon-name="fa/xmark-large-regular"></gux-icon>
                </div>
              </button>
            </gux-button-slot>
          </template>
        </gux-dismiss-button>
        <h1 class="gux-modal-header" id="gux-i">
          <slot name="title"></slot>
        </h1>
        <div class="gux-modal-content">
          <p>
            <slot name="content"></slot>
          </p>
        </div>
        <div class="gux-button-footer">
          <div class="gux-start-align-buttons">
            <slot name="start-align-buttons"></slot>
          </div>
          <div class="gux-end-align-buttons">
            <slot name="end-align-buttons"></slot>
          </div>
        </div>
      </div>
    </dialog>
  </template>
  <div slot="title">
    Modal Title
  </div>
  <div slot="content">
    This contains the modal content.
  </div>
  <div slot="start-align-buttons">
    <gux-button-slot>
      <button>
        Cancel
      </button>
    </gux-button-slot>
  </div>
  <div slot="end-align-buttons">
    <gux-button-slot accent="primary">
      <button>
        Accept
      </button>
    </gux-button-slot>
  </div>
</gux-modal>
`;
